### DISCUSSION TITLE
`QueueDoesNotExist` vs `NonExistentQueue`

### DISCUSSION THREAD

**Discussion by dmeiri** [aws/aws-sdk-js-v3#3901] [Q&A] [answered]
Hi 👋🏼

I need to handle the error which occurs when trying to delete a queue but it does not exist.

I can do

```ts
import { QueueDoesNotExist } from '@aws-sdk/client-sqs';

// ...

catch(e) {
    if(e instanceof QueueDoesNotExist) {
        // Some error handling here...
    }
}
```

But error actually received is `AWS.SimpleQueueService.NonExistentQueue`

This error is not in `@aws-sdk/client-sqs` so I have to do something like this:

```ts
import { SQSServiceException } from '@aws-sdk/client-sqs';

// ...

catch(e) {
    if (e instanceof SQSServiceException && e.name === 'AWS.SimpleQueueService.NonExistentQueue') {
        // Some error handling here...
    }
}
```

These are the packages I use:

`"@aws-sdk/types": "3.162.0"`
`"@aws-sdk/client-sqs": "3.162.0"`

### Question 1

What is the difference between `NonExistentQueue` and `QueueDoesNotExist`?

### Question 2

Why `NonExistentQueue` is not in `@aws-sdk/client-sqs` like `QueueDoesNotExist`?

**Comment by aBurmeseDev** [comment]
Hi @dmeiri - thanks for reaching out.

If I understand this correctly, you're running into QueueDoesNotExist error as mentioned here in this post: https://repost.aws/knowledge-center/sqs-queuedoesnotexist-errors

These could be one of the possible causes:
- Check that the queue URL is correct
- Check to make sure correct region is set
- Check for a recently deleted queue

You can find the rest of possible causes in the blog post linked above.

Hope that helps!

**Comment by RanVaknin** [comment, marked as answer]
Hi @dmeiri ,

This was a modeling issue with the SQS API itself. [Related to](https://github.com/aws/aws-sdk/issues/105). 
You are using a really old SDK version, in the newer SDK versions we have changed the wire protocol to the new JSON protocol which SQS now supports, and this problem went away. 

You can update to the latest version by running:
`npm i @aws-sdk/client-sqs@latest`

Thanks,
Ran
